edsnlp/pipes/misc/measurements/patterns.py

```diff
@@ -1,3 +1,75 @@
+import itertools
+
+
+def update_config_per(configs: dict) -> dict:
+    output_configs = {}
+    for key, config in configs.items():
+        if "per" in config.keys():
+            per_config = {
+                "dim": config["dim"],
+                "degree": -1 * config["degree"],
+                "scale": 1 / config["scale"],
+                "terms": config["per"],
+                "followed_by": None,
+            }
+            config.pop("per")
+            output_configs.update({f"per_{key}": per_config})
+        output_configs.update({key: config})
+    return output_configs
+
+
+def get_scaled_terms(terms, scale_prefix):
+    """Generate scaled terms."""
+    return [f"{scale}{term}" for term, scale in itertools.product(terms, scale_prefix)]
+
+
+def create_scaled_config(key, config, scale_prefix, scale_value):
+    """Create a scaled configuration dictionary."""
+    if key.startswith("per_"):
+        scale_value = 1 / scale_value
+        scaled_key = f"per_{scale_prefix[0]}{key[4:]}"
+    else:
+        scaled_key = f"{scale_prefix[0]}{key}"
+
+    return {
+        scaled_key: {
+            "dim": config["dim"],
+            "degree": config["degree"],
+            "scale": scale_value,
```

Comment:
In the case of "liters", I believe this doesn't account for the fact that a liter is 1e-3 m3.
And while we have m3 -> 1, dm3 -> 1e-3, cm3 -> 1e-6, we have liter -> 1e-3, deciliter -> 1e-4, centiliter -> 1e-5.
Managing the scales is not easily done automatically...